script.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/sh
+
+initial_workload_ns=""
+pvc_name="dd-io-pvc-nf" # nf stands for noflag for dd
+unset pvc_snapshot_array
+unset pod_array
+unset pending_pod_array
+
+echo "Switching project to $initial_workload_ns"
+oc project $initial_workload_ns
+
+
+echo "Creating Snapshot"
+snapshot_array=()
+for i in {1..25}
+do
+    snapshot_name=$pvc_name-$i-snp-`date -u "+%Y-%m-%d-%H-%M-%S-%3N"`
+    snapshot_array+=("$snapshot_name")
+    echo "apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: $snapshot_name
+spec:
+  volumeSnapshotClassName: ocs-storagecluster-rbdplugin-snapclass
+  source:
+    persistentVolumeClaimName: $pvc_name-$i
+" | kubectl apply -f -
+sleep 1
+done
+
+sleep 10
+
+echo "Checking VolumeSnapshot status"
+pvc_snapshot_array=() # Array
+for i in "${snapshot_array[@]}"
+do
+    output=$(oc get volumesnapshots $i -o json |jq -r .status.readyToUse)
+    if [ "$output" == "true" ]
+    then
+        echo "$i is in Ready State"
+        pvc_snapshot_array+=("$i")
+    else
+        for timeout on {1..10}
+        do
+            output_time=$(oc get volumesnapshots $i -o json |jq -r .status.readyToUse)
+            if [ "$output_time" == "true" ]
+            then
+                echo "$i is in Ready State"
+                pvc_snapshot_array+=("$i")
+                break
+            else
+                sleep 10
+            fi
+        done
+    fi
+done
+
+
+echo "Creating PVC from Snapshot"
+for i in "${pvc_snapshot_array[@]}"
+do
+    echo "apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: $i-sp-pvc
+  labels:
+    snapshot: bz
+spec:
+  storageClassName: ocs-storagecluster-ceph-rbd
+  dataSource:
+    name: $i
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 50Gi
+" | kubectl apply -f -
+done
+
+sleep 30
+
+echo "Checking PVC status created from Snapshot"
+pod_array=()
+for i in "${pvc_snapshot_array[@]}"
+do
+    output=$(oc get pvc $i-sp-pvc -o json |jq -r .status.phase)
+    if [ "$output" == "Bound" ]
+    then
+        echo "$i is in Bound state"
+        pod_array+=("$i-sp-pvc")
+    else
+        for timeout on {1..15}
+        do
+            output_time=$(oc get pvc $i-sp-pvc -o json |jq -r .status.phase)
+            if [ "$output_time" == "Bound" ]
+            then
+                echo "$i-sp-pvc is in Bound state"
+                pod_array+=("$i-sp-pvc")
+                break
+            else
+                sleep 10
+            fi
+        done
+        echo "$i Failed to reach Bound State"            
```

Comment:
If you succeeded at line 100, we will still print an error here. A better scheme maybe printing the count in the array to state how many snaps/items are being processed for the next stage.

Reply:
i will remove the print state ?

Reply:
Done